aws/scripts/sh/gh_auth_login.sh

```
@@ -0,0 +1,47 @@
+#   exit_on_failure: Set to "true" to exit on authentication failure (default: true)
+#                   Set to "false" to continue execution on failure
+
+set -e
+
+EXIT_ON_FAILURE=${1:-true}
+
+echo "Retrieving GitHub token for authentication..."
+
+# Configure git to handle CodeBuild environment ownership issues
+git config --global --add safe.directory /codebuild/output/srcDownload/src 2>/dev/null || true
+git config --global --add safe.directory ${CODEBUILD_SRC_DIR} 2>/dev/null || true
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)